mounts/org.osbuild.ostree.deployment

```diff
@@ -32,6 +32,12 @@ SCHEMA_2 = """
     "type": "object",
     "required": ["deployment"],
     "properties": {
+      "source": {
+        "type": "string",
+        "pattern": "^(mount|tree)$",
+        "default": "tree",
+        "description": "The source of the OSTree filesystem tree"
+      },
```

Comment:
I think we should keep things explicit (and flexible) and require naming the mount here if mount is used as a URL (e.g. `mount://root/`).  It keeps it consistent with how it works in stages and makes sure there's no ambiguity with which mount will be used as the source if there are multiple, which can happen if a stage needs an ostree deployment from a disk but also needs to mount other partitions.

I guess it makes sense that an ostree deployment will never need more than one mount to deploy from, so requiring a single "primary" mount called `root` is probably fine, but it's also nice to have the flexibility to name mounts arbitrarily.  In `osbuild/images` for example, we always name mounts based on their mountpoint (in a manner similar to `systemd-escape`) to avoid conflicts (like `/` and `/root`).  It's a convenient rule to have to just be able to name a mount without needing to check if it's unique or the only one in a stage (or in this case, in a mount).

Reply:
How strongly do you feel about this?

Like you said I can't really think of a reason one would want two ostree mounts in a single stage, and also you kind of want the ostree mount to be on top of the root of the mounts so there's really only one place for it to go IIUC.

I was also trying to think about backwards compat here because right now this just defaults to `tree`, which is what the behavior was before this and I guess with a URL scheme we'd have to have it be `tree:///` but really if it's not a mount (or even if it is) there's really only one place for it to go. Either `/run/osbuild/mounts` or `/run/osbuild/tree`.

Reply:
I'm quite strongly in favour of the schema being consistent with mounts.  It makes manifest generation code simpler.  The arbitrary naming is a bonus in both explicitness and flexibility.

Changing the default value shouldn't be an issue because there was never a value to begin with, it just always worked on the pipeline tree and now we're adding the option to work on a mount instead.  How we refer to the tree is up to the schema of the stage options, and generally (in every other stage where we need to refer to the tree or a subpath) that's `tree:///` (for the root of the tree).

Reply:
OK I'll look into making this change.

Reply:
OK - spent some time trying to do this today ultimately came to realize the API for a mount is different than a stage:

```
mount/root (org.osbuild.xfs): mounting /dev/loop0p4 -> /var/osbuild/store/tmp/buildroot-tmp-gn03xqf4/mounts/
mount/ostree.deployment (org.osbuild.ostree.deployment): {'options': {'deployment': {'osname': 'fedora-coreos', 'ref': 'ostree/1/1/0'},
mount/ostree.deployment (org.osbuild.ostree.deployment):              'source': 'mount://root/'},
mount/ostree.deployment (org.osbuild.ostree.deployment):  'root': '/var/osbuild/store/tmp/buildroot-tmp-gn03xqf4/mounts',
mount/ostree.deployment (org.osbuild.ostree.deployment):  'source': None,
mount/ostree.deployment (org.osbuild.ostree.deployment):  'target': None,
mount/ostree.deployment (org.osbuild.ostree.deployment):  'tree': '/var/osbuild/store/stage/uuid-d9e6ca87bb014711b0c0229c82968703/data/tree'}
```

so I can't really look into `args.["mounts"]` to find the path for a given mount. We just know where the root of all mount points are in `args["root"]`.

Reply:
(talked about this in chat; summarising here)
Mounts don't know about other mounts so my idea doesn't work.  I think in that case we can leave it as is and make sure it's documented how it should work in the description of the option.